.github/workflows/package-check.yml

```diff
@@ -0,0 +1,48 @@
+name: package-check
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  package-check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Use Node.js 20.x
+        uses: actions/setup-node@v4.2.0
+        with:
+          cache: 'yarn'
+          node-version: '20.x'
+
+      - name: install dependencies
+        run: yarn install
+
+      - name: rebuild the dist/ directory
+        run: yarn build && yarn package
```

Comment:
what if something like it ?
`yarn install && yarn build && yarn format && yarn package` same as in https://github.com/tj-actions/changed-files/blob/main/package.json#L18C42-L18C51 and it is called there https://github.com/tj-actions/changed-files/blob/main/.github/workflows/test.yml#L72 
```
      - run: yarn install && yarn build && yarn format && yarn package
      - run: git status --porcelain
      - run: git diff --exit-code
```

Reply:
Yep that could work as well. I'm pretty much going to leave this PR here "*as-is*" since it works and this is an implementation I have personally used in the past and it comes right from GitHub's own public template on building Actions. If the maintainers want to adopt it, tweak it, or make modifications that is totally fine and I would expect them to do so.

The extra reasons why I chose this approach was:

- It provides a way to debug unexpected (or even malicious) changes via the `actions/upload-artifact` step
- Calling a `yarn format` (or similar step) hasn't ever been something that is required when I used this in the past

Reply:
Sounds good!
It will help to catch the similar issue in the future, no matter which implementation to use.
Thank you for creating the PR @GrantBirki